src/sync/arc.rs

```diff
@@ -140,7 +141,7 @@ impl<T> ops::Deref for Arc<T> {
 
 impl<T> Clone for Arc<T> {
     fn clone(&self) -> Arc<T> {
-        self.inner.obj.ref_inc();
+        self.inner.obj.ref_inc(location!());
```

Comment:
do you think we should add `#[track_caller]` attributes to the `Clone` and `Drop` impls, too, so that we record the location in the _user_ code where the `Arc` was cloned or dropped from?